include/fmt/format.h

```diff
@@ -3833,18 +3833,28 @@ FMT_CONSTEXPR void advance_to(
     auto s = fmt::format("{}", fmt::ptr(p));
   \endrst
  */
-template <typename T> inline const void* ptr(const T* p) { return p; }
-template <typename T> inline const void* ptr(const std::unique_ptr<T>& p) {
+template <typename T>
+#if FMT_CLANG_VERSION && FMT_CLANG_VERSION < 700
+enable_if_t<!std::is_function<T>::value, const void*>
+#else
+const void*
+#endif
+ptr(const T* p) {
+  return p;
+}
```

Comment:
Can we change this to

```c++
template <typename T>
const void* ptr(T p) {
  static_assert(std::is_pointer<T>::value, "");
  return detail::bit_cast<const void*>(p);
}
```
and remove the `ptr(T (*fn)(Args...))` overload below?

Reply:
Yes, we can. Another great code simplification.